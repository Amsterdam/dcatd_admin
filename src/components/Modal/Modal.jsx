import React from 'react';
import PropTypes from 'prop-types';
import { Modal as SemanticModal } from 'semantic-ui-react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { setModal } from '../../actions/modal/modal';

import './modal.scss';

const mapStateToProps = state => ({
  modal: state.modal
});

const mapDispatchToProps = dispatch => bindActionCreators({
  setModal
}, dispatch);

const modalHeader = {
  AUTH_ERROR: 'Niet bevoegd',
  SERVER_ERROR: 'Er ging iets mis'
};

const modalContent = {
  AUTH_ERROR: `
    <span>
      <p>U bent niet bevoegd tot het beschikbaar maken van een dataset.</p>
    </span>
  `,
  SERVER_ERROR: `
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      <p>Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
      </p>
    </span>
  `
};

const Modal = props => (
  <SemanticModal
    open={props.modal.open}
    size="tiny"
  >
    <SemanticModal.Header>
      { modalHeader[props.modal.content] || 'Let op!' }
    </SemanticModal.Header>
    <SemanticModal.Content>
      <div dangerouslySetInnerHTML={{
        __html: modalContent[props.modal.content] || props.modal.content
      }}
      />
    </SemanticModal.Content>
    <SemanticModal.Actions>
      <button
        onClick={() => {
          props.setModal({ open: false });
          props.modal.onProceed();
        }}
        className="dcatd-form-button dcatd-form-button-submit"
      >
        {props.modal.actionLabel}
      </button>
      {props.modal.cancelLabel ?
        <button
          onClick={() => props.setModal({ open: false })}
          className="dcatd-form-button dcatd-form-button-cancel"
        >
          {props.modal.cancelLabel}
        </button>
        : ''}
    </SemanticModal.Actions>
  </SemanticModal>
);

Modal.defaultProps = {
  open: false,
  modal: {
    content: 'Weet u het zeker?',
    actionLabel: 'OK',
    cancelLabel: 'Annuleren',
    open: false,
    onProceed: () => {}
  },

  setModal: () => {}
};

Modal.propTypes = {
  modal: PropTypes.object,
  setModal: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Modal);
