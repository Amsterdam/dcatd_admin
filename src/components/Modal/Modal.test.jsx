import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import Modal from './Modal';

const middlewares = [thunk];

describe('The Modal component', () => {
  let store;
  let wrap;

  beforeEach(() => {
    store = configureMockStore(middlewares)();
    wrap = shallow(<Modal />, { context: { store } }).dive();
  });

  // TODO This fails
  // describe('renders component', () => {
  //   it('renders default props', () => {
  //     expect(wrap).toMatchSnapshot();
  //   });

  //   it('renders SERVER_ERROR', () => {
  //     wrap.setProps({
  //       modal: {
  //         actionLabel: 'OK',
  //         cancelLabel: null,
  //         content: 'SERVER_ERROR',
  //         onProceed: expect.any(Function),
  //         open: true
  //       }
  //     });

  //     expect(wrap).toMatchSnapshot();
  //   });
  // });

  describe('triggering events', () => {
    let setModalSpy;
    let onProceedSpy;

    beforeEach(() => {
      setModalSpy = jest.fn();
      onProceedSpy = jest.fn();
    });

    it('clicking submit closes modal and triggers onProceed', () => {
      wrap.setProps({
        modal: {
          onProceed: onProceedSpy
        },
        setModal: setModalSpy
      });

      wrap.find('.dcatd-form-button-submit').simulate('click');

      expect(onProceedSpy).toHaveBeenCalled();
      expect(setModalSpy).toHaveBeenCalledWith({ open: false });
    });

    it('clicking cancel closes modal', () => {
      wrap.setProps({
        setModal: setModalSpy
      });

      wrap.find('.dcatd-form-button-cancel').simulate('click');

      expect(onProceedSpy).not.toHaveBeenCalled();
      expect(setModalSpy).toHaveBeenCalledWith({ open: false });
    });
  });
});
