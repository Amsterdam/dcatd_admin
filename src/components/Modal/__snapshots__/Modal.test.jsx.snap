// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Modal component renders component renders SERVER_ERROR 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    modal={
      Object {
        "actionLabel": "OK",
        "cancelLabel": null,
        "content": "SERVER_ERROR",
        "onProceed": Any {
          "$$typeof": Symbol(jest.asymmetricMatcher),
          "sample": [Function],
        },
        "open": true,
      }
    }
    open={false}
    setModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ModalHeader>
          Er ging iets mis
        </ModalHeader>,
        <ModalContent>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
              }
            }
          />
        </ModalContent>,
        <ModalActions>
          <button
            className="dcatd-form-button dcatd-form-button-submit"
            onClick={[Function]}
          >
            OK
          </button>
          
        </ModalActions>,
      ],
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "open": true,
      "size": "tiny",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Er ging iets mis",
        },
        "ref": null,
        "rendered": "Er ging iets mis",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              onClick={[Function]}
            >
              OK
            </button>,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OK",
              "className": "dcatd-form-button dcatd-form-button-submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": "button",
          },
          "",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ModalHeader>
            Er ging iets mis
          </ModalHeader>,
          <ModalContent>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
                }
              }
            />
          </ModalContent>,
          <ModalActions>
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              onClick={[Function]}
            >
              OK
            </button>
            
          </ModalActions>,
        ],
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "open": true,
        "size": "tiny",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Er ging iets mis",
          },
          "ref": null,
          "rendered": "Er ging iets mis",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "dangerouslySetInnerHTML": Object {
                "__html": "
    <span>
      <p>U bent niet (meer) bevoegd tot het uitvoeren van deze actie.</p>
      Wellicht komt dit doordat:
      <ol>
        <li>Uw sessie is verlopen. Deze duurt maximaal 10 uur.</li>
        <li>U niet bevoegd bent om datasets te beheren.</li>
      </ol>
    </span>
  ",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <button
                className="dcatd-form-button dcatd-form-button-submit"
                onClick={[Function]}
              >
                OK
              </button>,
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OK",
                "className": "dcatd-form-button dcatd-form-button-submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "OK",
              "type": "button",
            },
            "",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`The Modal component renders component renders default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    modal={
      Object {
        "actionLabel": "OK",
        "cancelLabel": "Annuleren",
        "content": "Weet u het zeker?",
        "onProceed": [Function],
        "open": false,
      }
    }
    open={false}
    setModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ModalHeader />,
        <ModalContent>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />
        </ModalContent>,
        <ModalActions>
          <button
            className="dcatd-form-button dcatd-form-button-submit"
            onClick={[Function]}
          >
            OK
          </button>
          <button
            className="dcatd-form-button dcatd-form-button-cancel"
            onClick={[Function]}
          >
            Annuleren
          </button>
        </ModalActions>,
      ],
      "closeOnDimmerClick": true,
      "closeOnDocumentClick": false,
      "dimmer": true,
      "eventPool": "Modal",
      "open": false,
      "size": "tiny",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "dangerouslySetInnerHTML": Object {
              "__html": undefined,
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              onClick={[Function]}
            >
              OK
            </button>,
            <button
              className="dcatd-form-button dcatd-form-button-cancel"
              onClick={[Function]}
            >
              Annuleren
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OK",
              "className": "dcatd-form-button dcatd-form-button-submit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "OK",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Annuleren",
              "className": "dcatd-form-button dcatd-form-button-cancel",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Annuleren",
            "type": "button",
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ModalHeader />,
          <ModalContent>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
            />
          </ModalContent>,
          <ModalActions>
            <button
              className="dcatd-form-button dcatd-form-button-submit"
              onClick={[Function]}
            >
              OK
            </button>
            <button
              className="dcatd-form-button dcatd-form-button-cancel"
              onClick={[Function]}
            >
              Annuleren
            </button>
          </ModalActions>,
        ],
        "closeOnDimmerClick": true,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "open": false,
        "size": "tiny",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "dangerouslySetInnerHTML": Object {
                "__html": undefined,
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <button
                className="dcatd-form-button dcatd-form-button-submit"
                onClick={[Function]}
              >
                OK
              </button>,
              <button
                className="dcatd-form-button dcatd-form-button-cancel"
                onClick={[Function]}
              >
                Annuleren
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OK",
                "className": "dcatd-form-button dcatd-form-button-submit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "OK",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Annuleren",
                "className": "dcatd-form-button dcatd-form-button-cancel",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Annuleren",
              "type": "button",
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
